### PR TITLE
Refs [TASK][36338] : Create Calculator Screen

### PR DESCRIPTION
**Pull request by hideoncalm**
## Related Tickets
- [#Task 36338: Create Calculator Screen](https://edu-redmine.sun-asterisk.vn/issues/36338)
## WHAT
- 
## Evidence (Screenshot or Video)
image

## Review Checklist

Category | View Point | Description | Self review | Reviewer2 (name)
--- | --- | --- | --- | ---
Conventions | Sun* coding conventions | https://github.com/framgia/coding-standards/tree/master/eng/android |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
Conventions | Kotlin coding conventions | https://kotlinlang.org/docs/coding-conventions.html |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
Redmine | Sun* Redmine working process  | https://github.com/framgia/Training-Guideline/blob/master/WorkingProcess/redmine/redmine.md |<li>- [ ] yes</li>|<li>- [ ] yes</li>  

## Notes (Optional)
*(Impacted Areas in Application(List features, api, models or services that this PR will affect))*

*(List gem, library third party add new)*

*(Other notes)*

### CODE REVIEW

File: app/src/main/java/com/sunasterisk/coinqapp/ui/calculator/CalculatorFragment.kt [hunk 1]
```diff
@@ -1,15 +1,101 @@
...
+            buttonBack.setOnClickListener {
+                backPress(parentFragmentManager, SettingFragment())
+            }
+            editTextSearch.setOnEditorActionListener(onEditTextListener())
+        }
+    }
+
+    override fun showCoins(coins: List<Coin>) {
+        coinsData = coins.toMutableList()
+        adapter.updateData(coinsData)
```
Comment (thonc-2135) [inline app/src/main/java/com/sunasterisk/coinqapp/ui/calculator/CalculatorFragment.kt]:
```suggestion
        adapter.updateData(coins.toMutableList())
```

File: app/src/main/java/com/sunasterisk/coinqapp/ui/calculator/CalculatorFragment.kt [hunk 2]
```diff
@@ -1,15 +1,101 @@
...
+
+    private fun onEditTextListener() = TextView.OnEditorActionListener { _, _, _ ->
+        searchListener(binding.editTextSearch)
+        true
+    }
+
+    private fun searchListener(editText: EditText) {
+        val key = editText.text.toString()
+        if (key.isEmpty()) {
+            context?.showMessage("Fill Search")
```
Comment (thonc-2135) [inline app/src/main/java/com/sunasterisk/coinqapp/ui/calculator/CalculatorFragment.kt]: hard code

File: app/src/main/java/com/sunasterisk/coinqapp/ui/calculator/CalculatorFragment.kt [hunk 3]
```diff
@@ -1,15 +1,101 @@
...
+        } else {
+            adapter = CalculatorAdapter(::onItemClick, key.toDouble())
+            adapter.updateData(coinsData)
+            binding.recyclerCoins.adapter = adapter
```
Comment (thonc-2135) [inline app/src/main/java/com/sunasterisk/coinqapp/ui/calculator/CalculatorFragment.kt]: lần nào cx đi tạo adapter ms ư ???

Comment (hideoncalm) [reply app/src/main/java/com/sunasterisk/coinqapp/ui/calculator/CalculatorFragment.kt]: Vâng vì cái adapter dữ liệu trong adapter nó thay đổi sau mỗi lần em nhập edit text ạ

File: app/src/main/java/com/sunasterisk/coinqapp/ui/calculator/CalculatorFragment.kt (hunk 1 above)
Comment (hideoncalm) [reply app/src/main/java/com/sunasterisk/coinqapp/ui/calculator/CalculatorFragment.kt]: em tạo mảng coinsData để lưu lại cái coins sau khi gọi api ạ

File: app/src/main/java/com/sunasterisk/coinqapp/ui/calculator/adapter/CalculatorAdapter.kt [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+                false
+            ),
+            onItemClick
+        )
+
+    fun updateNewData(newData: MutableList<Coin>, numberUSD: Double) {
+        val data = newData.mapIndexed { _, coin ->
+            coin.copy(price = numberUSD/(coin.price))
+        }
+        updateData(data as MutableList<Coin>)
```
Comment (thonc-2135) [inline app/src/main/java/com/sunasterisk/coinqapp/ui/calculator/adapter/CalculatorAdapter.kt]:
```suggestion
        updateData(data.toMutableList())
```

File: app/src/main/java/com/sunasterisk/coinqapp/ui/calculator/CalculatorFragment.kt [hunk 5]
```diff
@@ -1,15 +1,99 @@
...
+    private val loadingProgressBar by lazy { CustomProgressBar() }
+
+    override val binding by lazy {
+        FragmentCalculatorBinding.inflate(layoutInflater)
+    }
 
     override fun initViews() {
+        with(binding) {
+            recyclerCoins.adapter = adapter
+        }
```
Comment (huytq-2422) [inline app/src/main/java/com/sunasterisk/coinqapp/ui/calculator/CalculatorFragment.kt]:
```suggestion
            binding.recyclerCoins.adapter = adapter
```

**Pull request merged**